core/arch/arm/tee/tee_rpmb.c

```diff
@@ -431,22 +503,21 @@ static TEE_Result tee_rpmb_req_pack(struct rpmb_req *req,
 			u16_to_bytes(*rawdata->blk_idx, datafrm[i].address);
 		}
 
-		if (rawdata->write_counter != NULL)
+		if (rawdata->write_counter)
 			u32_to_bytes(*rawdata->write_counter,
 				     datafrm[i].write_counter);
 
-		if (rawdata->nonce != NULL)
+		if (rawdata->nonce)
 			memcpy(datafrm[i].nonce, rawdata->nonce,
 			       RPMB_NONCE_SIZE);
 
-		if (rawdata->data != NULL)
-			for (j = 0; j < nbr_frms; j++)
-				memcpy((datafrm + j)->data,
-				       rawdata->data + RPMB_DATA_SIZE * j,
-				       RPMB_DATA_SIZE);
+		if (rawdata->data)
+			memcpy(datafrm[i].data,
+			       rawdata->data + j,
```

Comment:
replacing `j` with `i * RPMB_DATA_SIZE` would make the code easier as we could remove one index variable.